.github/workflows/e2e-tests.yml

```diff
@@ -12,15 +12,15 @@ on:
         type: string
 
 jobs:
-
   e2e:
     name: Run E2E tests (${{ matrix.shard }} of ${{ matrix.total_shards }})
     runs-on: ubuntu-latest
 
     strategy:
-        matrix:
-          shard: [1, 2, 3]
-          total_shards: [3] # Github Actions doesn't have a built-in method to get the length of an array
+      fail-fast: false # Allows e2e test report to be merged if only some of the shards have passed
```

Comment:
Nit: I think a comment like: "Continue running tests to completion even if some shards fail" makes slightly more sense to me, since fail-fast: true wouldn't just prevent reports from merging it would also stop tests early.